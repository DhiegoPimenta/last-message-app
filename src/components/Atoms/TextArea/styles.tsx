import styled, { css } from 'styled-components'
import { TextAreaProps } from '.'

export const TextArea = styled.textarea<TextAreaProps>`
    ${({ theme }) => css`
        border: 4px solid ${theme.colors.danger};
        border-radius: ${theme.border.radius};
        width: 100%;
        height: 33rem;

        ::placeholder {
            color: ${theme.colors.gray};
            font-size: ${theme.font.sizes.small};
            padding: 3rem;
            font-weight: ${theme.font.bold};
        }
    `}
`
