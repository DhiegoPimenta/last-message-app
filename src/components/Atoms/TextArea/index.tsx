import * as S from './styles'

export type TextAreaProps = any

const TextArea = (props: TextAreaProps) => <S.TextArea {...props} />

export default TextArea
