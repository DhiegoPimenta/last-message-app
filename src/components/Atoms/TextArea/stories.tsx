import { Story, Meta } from '@storybook/react/types-6-0'

import TextArea from '.'

export default {
    title: 'Atoms/TextArea',
    component: TextArea,
} as Meta

export const Basic: Story = (args) => <TextArea {...args} />
