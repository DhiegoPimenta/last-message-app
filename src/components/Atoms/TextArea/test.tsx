import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import TextArea from '.'

describe('<TextArea />', () => {
    it('Should render the textarea', () => {
        const { container } = renderWithTheme(<TextArea />)

        expect(
            screen.getByLabelText(/campo de inserção de texto/i)
        ).toBeInTheDocument()

        expect(container.firstChild).toMatchSnapshot()
    })
})
