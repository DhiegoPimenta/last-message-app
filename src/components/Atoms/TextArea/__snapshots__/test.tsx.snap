// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> Should render the textarea 1`] = `
.c0 {
  border: 4px solid #FBA9B0;
  border-radius: 1rem;
  width: 100%;
  height: 33rem;
}

.c0::-webkit-input-placeholder {
  color: #CECECE;
  font-size: 1.8rem;
  padding: 3rem;
  font-weight: 600;
}

.c0::-moz-placeholder {
  color: #CECECE;
  font-size: 1.8rem;
  padding: 3rem;
  font-weight: 600;
}

.c0:-ms-input-placeholder {
  color: #CECECE;
  font-size: 1.8rem;
  padding: 3rem;
  font-weight: 600;
}

.c0::placeholder {
  color: #CECECE;
  font-size: 1.8rem;
  padding: 3rem;
  font-weight: 600;
}

<textarea
  aria-label="campo de inserção de texto"
  class="c0"
  placeholder="Escreva aqui sua mensagem..."
/>
`;
